### ISSUE TITLE
Tried to compile on Linux Mint but got errors...

### ISSUE THREAD

**Issue by vasagle-gleblu** [BatchDrake/sigutils#43]
Hello,
I am trying to build this package so that I can run SigDigger on my linux laptop.  However, when I try to run the make command (2nd step) I receive an error.  Not sure what I'm doing wrong.  Any help is appreciated!

~/Projects/sigutils/build$ make
[ 63%] Built target sigutils
[ 63%] Linking C executable sutest
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `sf_flac_meta_get_vorbiscomments':
flac.c:(.text+0xdd6): undefined reference to `FLAC__metadata_object_vorbiscomment_find_entry_from'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `sf_flac_error_callback':
flac.c:(.text+0x1259): undefined reference to `FLAC__StreamDecoderErrorStatusString'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_write_strings':
flac.c:(.text+0x1444): undefined reference to `FLAC__metadata_object_new'
/usr/bin/ld: flac.c:(.text+0x1591): undefined reference to `FLAC__metadata_object_vorbiscomment_entry_from_name_value_pair'
/usr/bin/ld: flac.c:(.text+0x15ad): undefined reference to `FLAC__metadata_object_vorbiscomment_append_comment'
/usr/bin/ld: flac.c:(.text+0x15fc): undefined reference to `FLAC__stream_encoder_set_metadata'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_write_header':
flac.c:(.text+0x16a0): undefined reference to `FLAC__stream_encoder_init_stream'
/usr/bin/ld: flac.c:(.text+0x16bd): undefined reference to `FLAC__StreamEncoderInitStatusString'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_close':
flac.c:(.text+0x1944): undefined reference to `FLAC__metadata_object_delete'
/usr/bin/ld: flac.c:(.text+0x1963): undefined reference to `FLAC__stream_encoder_finish'
/usr/bin/ld: flac.c:(.text+0x1973): undefined reference to `FLAC__stream_encoder_delete'
/usr/bin/ld: flac.c:(.text+0x19a4): undefined reference to `FLAC__stream_decoder_finish'
/usr/bin/ld: flac.c:(.text+0x19b3): undefined reference to `FLAC__stream_decoder_delete'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_enc_init':
flac.c:(.text+0x1ae5): undefined reference to `FLAC__stream_encoder_delete'
/usr/bin/ld: flac.c:(.text+0x1aea): undefined reference to `FLAC__stream_encoder_new'
/usr/bin/ld: flac.c:(.text+0x1b27): undefined reference to `FLAC__stream_encoder_set_channels'
/usr/bin/ld: flac.c:(.text+0x1b75): undefined reference to `FLAC__stream_encoder_set_sample_rate'
/usr/bin/ld: flac.c:(.text+0x1bba): undefined reference to `FLAC__stream_encoder_set_bits_per_sample'
/usr/bin/ld: flac.c:(.text+0x1bfc): undefined reference to `FLAC__stream_encoder_set_compression_level'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_read_header':
flac.c:(.text+0x1c8f): undefined reference to `FLAC__stream_decoder_delete'
/usr/bin/ld: flac.c:(.text+0x1c94): undefined reference to `FLAC__stream_decoder_new'
/usr/bin/ld: flac.c:(.text+0x1cc0): undefined reference to `FLAC__stream_decoder_set_metadata_respond_all'
/usr/bin/ld: flac.c:(.text+0x1d0d): undefined reference to `FLAC__stream_decoder_init_stream'
/usr/bin/ld: flac.c:(.text+0x1d2b): undefined reference to `FLAC__stream_decoder_process_until_end_of_metadata'
/usr/bin/ld: flac.c:(.text+0x1d60): undefined reference to `FLAC__stream_decoder_delete'
/usr/bin/ld: flac.c:(.text+0x1d78): undefined reference to `FLAC__stream_decoder_get_decode_position'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_read_loop':
flac.c:(.text+0x2084): undefined reference to `FLAC__stream_decoder_get_state'
/usr/bin/ld: flac.c:(.text+0x209f): undefined reference to `FLAC__StreamDecoderStateString'
/usr/bin/ld: flac.c:(.text+0x211d): undefined reference to `FLAC__stream_decoder_process_single'
/usr/bin/ld: flac.c:(.text+0x2159): undefined reference to `FLAC__stream_decoder_get_state'
/usr/bin/ld: flac.c:(.text+0x2174): undefined reference to `FLAC__StreamDecoderStateString'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_write_s2flac':
flac.c:(.text+0x25d8): undefined reference to `FLAC__stream_encoder_process_interleaved'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_write_i2flac':
flac.c:(.text+0x2747): undefined reference to `FLAC__stream_encoder_process_interleaved'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_write_f2flac':
flac.c:(.text+0x2906): undefined reference to `FLAC__stream_encoder_process_interleaved'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_write_d2flac':
flac.c:(.text+0x2f5e): undefined reference to `FLAC__stream_encoder_process_interleaved'
/usr/bin/ld: /usr/local/lib/libsndfile.a(flac.c.o): in function `flac_seek':
flac.c:(.text+0x34d2): undefined reference to `FLAC__stream_decoder_seek_absolute'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_dec_close':
mpeg_decode.c:(.text+0x122): undefined reference to `mpg123_close'
/usr/bin/ld: mpeg_decode.c:(.text+0x131): undefined reference to `mpg123_delete'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_dec_decode':
mpeg_decode.c:(.text+0x1c4): undefined reference to `mpg123_read'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_dec_seek':
mpeg_decode.c:(.text+0x7a3): undefined reference to `mpg123_seek'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_dec_fill_sfinfo':
mpeg_decode.c:(.text+0x7f8): undefined reference to `mpg123_getformat'
/usr/bin/ld: mpeg_decode.c:(.text+0x82c): undefined reference to `mpg123_length'
/usr/bin/ld: mpeg_decode.c:(.text+0x88d): undefined reference to `mpg123_format'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_decoder_read_strings':
mpeg_decode.c:(.text+0x1420): undefined reference to `mpg123_id3'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_dec_byterate':
mpeg_decode.c:(.text+0x14b9): undefined reference to `mpg123_info'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_decoder_init':
mpeg_decode.c:(.text+0x152a): undefined reference to `mpg123_init'
/usr/bin/ld: mpeg_decode.c:(.text+0x1585): undefined reference to `mpg123_new'
/usr/bin/ld: mpeg_decode.c:(.text+0x15a2): undefined reference to `mpg123_plain_strerror'
/usr/bin/ld: mpeg_decode.c:(.text+0x15fb): undefined reference to `mpg123_replace_reader_handle'
/usr/bin/ld: mpeg_decode.c:(.text+0x1620): undefined reference to `mpg123_param'
/usr/bin/ld: mpeg_decode.c:(.text+0x1645): undefined reference to `mpg123_param'
/usr/bin/ld: mpeg_decode.c:(.text+0x16d0): undefined reference to `mpg123_param'
/usr/bin/ld: mpeg_decode.c:(.text+0x174d): undefined reference to `mpg123_open_handle'
/usr/bin/ld: mpeg_decode.c:(.text+0x1761): undefined reference to `mpg123_plain_strerror'
/usr/bin/ld: mpeg_decode.c:(.text+0x17b1): undefined reference to `mpg123_plain_strerror'
/usr/bin/ld: mpeg_decode.c:(.text+0x17ec): undefined reference to `mpg123_info'
/usr/bin/ld: mpeg_decode.c:(.text+0x1800): undefined reference to `mpg123_plain_strerror'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_decode.c.o): in function `mpeg_decoder_get_bitrate_mode':
mpeg_decode.c:(.text+0x1980): undefined reference to `mpg123_info'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_init':
mpeg_l3_encode.c:(.text+0xa8): undefined reference to `lame_init'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xf1): undefined reference to `lame_set_in_samplerate'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x10c): undefined reference to `lame_set_num_channels'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x127): undefined reference to `lame_set_out_samplerate'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x149): undefined reference to `lame_set_write_id3tag_automatic'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x171): undefined reference to `lame_set_bWriteVbrTag'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_write_id3tag':
mpeg_l3_encode.c:(.text+0x2e2): undefined reference to `id3tag_init'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x373): undefined reference to `id3tag_set_title'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x3b7): undefined reference to `id3tag_set_artist'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x3fb): undefined reference to `id3tag_set_album'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x43f): undefined reference to `id3tag_set_year'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x483): undefined reference to `id3tag_set_comment'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x4c7): undefined reference to `id3tag_set_genre'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x508): undefined reference to `id3tag_set_track'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x532): undefined reference to `lame_get_id3v2_tag'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x592): undefined reference to `lame_get_id3v2_tag'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_set_quality':
mpeg_l3_encode.c:(.text+0x664): undefined reference to `lame_set_VBR_quality'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x718): undefined reference to `lame_set_VBR_mean_bitrate_kbps'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x731): undefined reference to `lame_set_brate'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_set_bitrate_mode':
mpeg_l3_encode.c:(.text+0x803): undefined reference to `lame_set_VBR'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_get_bitrate_mode':
mpeg_l3_encode.c:(.text+0x872): undefined reference to `lame_get_VBR'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_close':
mpeg_l3_encode.c:(.text+0x918): undefined reference to `lame_encode_flush'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x958): undefined reference to `lame_get_id3v1_tag'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x9ad): undefined reference to `lame_get_lametag_frame'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xa38): undefined reference to `lame_get_lametag_frame'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xaed): undefined reference to `lame_close'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_log_config':
mpeg_l3_encode.c:(.text+0xb1f): undefined reference to `lame_get_version'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xb73): undefined reference to `lame_get_mode'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xbe3): undefined reference to `lame_get_framesize'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xc25): undefined reference to `lame_get_out_samplerate'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xc63): undefined reference to `lame_get_VBR'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xcb9): undefined reference to `lame_get_brate'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xcfc): undefined reference to `lame_get_VBR_mean_bitrate_kbps'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xd3c): undefined reference to `lame_get_VBR_q'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xd64): undefined reference to `lame_get_VBR'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xd8b): undefined reference to `lame_get_encoder_delay'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xdb1): undefined reference to `lame_get_bWriteVbrTag'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_construct':
mpeg_l3_encode.c:(.text+0xe0b): undefined reference to `lame_init_params'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0xe6e): undefined reference to `lame_get_framesize'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encoder_byterate':
mpeg_l3_encode.c:(.text+0xf33): undefined reference to `lame_get_brate'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_short_mono':
mpeg_l3_encode.c:(.text+0x1076): undefined reference to `lame_encode_buffer'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_short_stereo':
mpeg_l3_encode.c:(.text+0x1265): undefined reference to `lame_encode_buffer_interleaved'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_int_mono':
mpeg_l3_encode.c:(.text+0x140a): undefined reference to `lame_encode_buffer_int'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_int_stereo':
mpeg_l3_encode.c:(.text+0x1560): undefined reference to `lame_encode_buffer_interleaved_int'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_float_mono':
mpeg_l3_encode.c:(.text+0x16c0): undefined reference to `lame_encode_buffer_ieee_float'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x170a): undefined reference to `lame_encode_buffer_float'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_float_stereo':
mpeg_l3_encode.c:(.text+0x1947): undefined reference to `lame_encode_buffer_interleaved_ieee_float'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x19d6): undefined reference to `lame_encode_buffer_interleaved_ieee_float'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_double_mono':
mpeg_l3_encode.c:(.text+0x1c62): undefined reference to `lame_encode_buffer_ieee_double'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x1cef): undefined reference to `lame_encode_buffer_ieee_double'
/usr/bin/ld: /usr/local/lib/libsndfile.a(mpeg_l3_encode.c.o): in function `mpeg_l3_encode_write_double_stereo':
mpeg_l3_encode.c:(.text+0x1f25): undefined reference to `lame_encode_buffer_interleaved_ieee_double'
/usr/bin/ld: mpeg_l3_encode.c:(.text+0x1fb6): undefined reference to `lame_encode_buffer_interleaved_ieee_double'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_read_first_page':
ogg.c:(.text+0x95): undefined reference to `ogg_sync_buffer'
/usr/bin/ld: ogg.c:(.text+0xec): undefined reference to `ogg_sync_wrote'
/usr/bin/ld: ogg.c:(.text+0x164): undefined reference to `ogg_page_bos'
/usr/bin/ld: ogg.c:(.text+0x19c): undefined reference to `ogg_page_serialno'
/usr/bin/ld: ogg.c:(.text+0x1b0): undefined reference to `ogg_stream_reset_serialno'
/usr/bin/ld: ogg.c:(.text+0x1ce): undefined reference to `ogg_stream_pagein'
/usr/bin/ld: ogg.c:(.text+0x20f): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_sync_fseek':
ogg.c:(.text+0x39b): undefined reference to `ogg_sync_reset'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_sync_next_page':
ogg.c:(.text+0x3f3): undefined reference to `ogg_sync_pageseek'
/usr/bin/ld: ogg.c:(.text+0x4cc): undefined reference to `ogg_sync_buffer'
/usr/bin/ld: ogg.c:(.text+0x53f): undefined reference to `ogg_sync_wrote'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_stream_next_page':
ogg.c:(.text+0x5fd): undefined reference to `ogg_page_serialno'
/usr/bin/ld: ogg.c:(.text+0x628): undefined reference to `ogg_page_eos'
/usr/bin/ld: ogg.c:(.text+0x658): undefined reference to `ogg_stream_pagein'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_stream_unpack_page':
ogg.c:(.text+0x6c6): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: ogg.c:(.text+0x71a): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: ogg.c:(.text+0x767): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: ogg.c:(.text+0x796): undefined reference to `ogg_stream_packetpeek'
/usr/bin/ld: ogg.c:(.text+0x7cc): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_sync_last_page_before':
ogg.c:(.text+0x903): undefined reference to `ogg_page_serialno'
/usr/bin/ld: ogg.c:(.text+0x91a): undefined reference to `ogg_page_granulepos'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_stream_seek_page_search':
ogg.c:(.text+0xaa5): undefined reference to `ogg_stream_reset_serialno'
/usr/bin/ld: ogg.c:(.text+0xbdf): undefined reference to `ogg_stream_reset'
/usr/bin/ld: ogg.c:(.text+0xcda): undefined reference to `ogg_stream_reset'
/usr/bin/ld: ogg.c:(.text+0xda1): undefined reference to `ogg_page_serialno'
/usr/bin/ld: ogg.c:(.text+0xdb9): undefined reference to `ogg_page_packets'
/usr/bin/ld: ogg.c:(.text+0xddc): undefined reference to `ogg_page_granulepos'
/usr/bin/ld: ogg.c:(.text+0xe20): undefined reference to `ogg_stream_pagein'
/usr/bin/ld: ogg.c:(.text+0xe38): undefined reference to `ogg_stream_reset'
/usr/bin/ld: ogg.c:(.text+0xeb1): undefined reference to `ogg_stream_reset'
/usr/bin/ld: ogg.c:(.text+0x1098): undefined reference to `ogg_stream_pagein'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_open':
ogg.c:(.text+0x11cf): undefined reference to `ogg_sync_clear'
/usr/bin/ld: ogg.c:(.text+0x11df): undefined reference to `ogg_stream_clear'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_close':
ogg.c:(.text+0x12a2): undefined reference to `ogg_sync_clear'
/usr/bin/ld: ogg.c:(.text+0x12b2): undefined reference to `ogg_stream_clear'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_stream_classify':
ogg.c:(.text+0x12d9): undefined reference to `ogg_sync_init'
/usr/bin/ld: ogg.c:(.text+0x12ee): undefined reference to `ogg_stream_init'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_page_classify':
ogg.c:(.text+0x14ff): undefined reference to `ogg_page_serialno'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg.c.o): in function `ogg_page_search_continued_data':
ogg.c:(.text+0x1779): undefined reference to `ogg_stream_pagein'
/usr/bin/ld: ogg.c:(.text+0x1798): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: /usr/local/lib/libsndfile.a(id3.c.o): in function `id3_lookup_v1_genre':
id3.c:(.text+0x70): undefined reference to `id3tag_genre_list'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_read_header':
ogg_vorbis.c:(.text+0x72): undefined reference to `vorbis_info_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0x82): undefined reference to `vorbis_comment_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0xc9): undefined reference to `ogg_stream_packetpeek'
/usr/bin/ld: ogg_vorbis.c:(.text+0x115): undefined reference to `vorbis_synthesis_headerin'
/usr/bin/ld: ogg_vorbis.c:(.text+0x165): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: ogg_vorbis.c:(.text+0x227): undefined reference to `vorbis_synthesis_headerin'
/usr/bin/ld: ogg_vorbis.c:(.text+0x259): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: ogg_vorbis.c:(.text+0x3bd): undefined reference to `ogg_page_eos'
/usr/bin/ld: ogg_vorbis.c:(.text+0x526): undefined reference to `vorbis_comment_query'
/usr/bin/ld: ogg_vorbis.c:(.text+0x673): undefined reference to `vorbis_synthesis_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0x691): undefined reference to `vorbis_block_init'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_write_header':
ogg_vorbis.c:(.text+0x700): undefined reference to `vorbis_info_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0x748): undefined reference to `vorbis_encode_init_vbr'
/usr/bin/ld: ogg_vorbis.c:(.text+0x782): undefined reference to `vorbis_comment_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0x7a3): undefined reference to `vorbis_comment_add_tag'
/usr/bin/ld: ogg_vorbis.c:(.text+0x91b): undefined reference to `vorbis_comment_add_tag'
/usr/bin/ld: ogg_vorbis.c:(.text+0x956): undefined reference to `vorbis_analysis_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0x97a): undefined reference to `vorbis_block_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0x996): undefined reference to `ogg_stream_init'
/usr/bin/ld: ogg_vorbis.c:(.text+0x9c9): undefined reference to `vorbis_analysis_headerout'
/usr/bin/ld: ogg_vorbis.c:(.text+0x9e6): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_vorbis.c:(.text+0xa00): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_vorbis.c:(.text+0xa1a): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_vorbis.c:(.text+0xa60): undefined reference to `ogg_stream_flush'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_close':
ogg_vorbis.c:(.text+0xb19): undefined reference to `vorbis_analysis_wrote'
/usr/bin/ld: ogg_vorbis.c:(.text+0xb35): undefined reference to `vorbis_analysis'
/usr/bin/ld: ogg_vorbis.c:(.text+0xb47): undefined reference to `vorbis_bitrate_addblock'
/usr/bin/ld: ogg_vorbis.c:(.text+0xb6a): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_vorbis.c:(.text+0xb8a): undefined reference to `ogg_stream_pageout'
/usr/bin/ld: ogg_vorbis.c:(.text+0xbc1): undefined reference to `ogg_page_eos'
/usr/bin/ld: ogg_vorbis.c:(.text+0xbff): undefined reference to `vorbis_bitrate_flushpacket'
/usr/bin/ld: ogg_vorbis.c:(.text+0xc25): undefined reference to `vorbis_analysis_blockout'
/usr/bin/ld: ogg_vorbis.c:(.text+0xc40): undefined reference to `vorbis_block_clear'
/usr/bin/ld: ogg_vorbis.c:(.text+0xc50): undefined reference to `vorbis_dsp_clear'
/usr/bin/ld: ogg_vorbis.c:(.text+0xc60): undefined reference to `vorbis_comment_clear'
/usr/bin/ld: ogg_vorbis.c:(.text+0xc70): undefined reference to `vorbis_info_clear'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `ogg_vorbis_open':
ogg_vorbis.c:(.text+0xd0d): undefined reference to `vorbis_version_string'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_read_sample':
ogg_vorbis.c:(.text+0x15b0): undefined reference to `vorbis_synthesis_read'
/usr/bin/ld: ogg_vorbis.c:(.text+0x15ed): undefined reference to `vorbis_synthesis_pcmout'
/usr/bin/ld: ogg_vorbis.c:(.text+0x164e): undefined reference to `vorbis_synthesis_restart'
/usr/bin/ld: ogg_vorbis.c:(.text+0x16a2): undefined reference to `vorbis_synthesis'
/usr/bin/ld: ogg_vorbis.c:(.text+0x16c4): undefined reference to `vorbis_synthesis_blockin'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_write_samples':
ogg_vorbis.c:(.text+0x1804): undefined reference to `vorbis_analysis_wrote'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1820): undefined reference to `vorbis_analysis'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1832): undefined reference to `vorbis_bitrate_addblock'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1855): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1875): undefined reference to `ogg_stream_pageout'
/usr/bin/ld: ogg_vorbis.c:(.text+0x18ac): undefined reference to `ogg_page_eos'
/usr/bin/ld: ogg_vorbis.c:(.text+0x18ea): undefined reference to `vorbis_bitrate_flushpacket'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1910): undefined reference to `vorbis_analysis_blockout'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_write_s':
ogg_vorbis.c:(.text+0x199d): undefined reference to `vorbis_analysis_buffer'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_write_i':
ogg_vorbis.c:(.text+0x1aac): undefined reference to `vorbis_analysis_buffer'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_write_f':
ogg_vorbis.c:(.text+0x1bbd): undefined reference to `vorbis_analysis_buffer'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_write_d':
ogg_vorbis.c:(.text+0x1cc0): undefined reference to `vorbis_analysis_buffer'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_skip':
ogg_vorbis.c:(.text+0x1da3): undefined reference to `vorbis_info_blocksize'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1dbb): undefined reference to `vorbis_synthesis_pcmout'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1e0a): undefined reference to `vorbis_synthesis_read'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1e9f): undefined reference to `vorbis_synthesis_restart'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1f10): undefined reference to `vorbis_packet_blocksize'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1fb7): undefined reference to `vorbis_synthesis_trackonly'
/usr/bin/ld: ogg_vorbis.c:(.text+0x1fd5): undefined reference to `vorbis_synthesis_blockin'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_seek_trysearch':
ogg_vorbis.c:(.text+0x20db): undefined reference to `vorbis_info_blocksize'
/usr/bin/ld: ogg_vorbis.c:(.text+0x21a0): undefined reference to `vorbis_synthesis_restart'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_seek':
ogg_vorbis.c:(.text+0x22ca): undefined reference to `ogg_stream_reset_serialno'
/usr/bin/ld: ogg_vorbis.c:(.text+0x2327): undefined reference to `vorbis_synthesis_restart'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_vorbis.c.o): in function `vorbis_calculate_granulepos':
ogg_vorbis.c:(.text+0x2453): undefined reference to `vorbis_packet_blocksize'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_open':
ogg_opus.c:(.text+0xfb): undefined reference to `opus_get_version_string'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_close':
ogg_opus.c:(.text+0x369): undefined reference to `ogg_packet_clear'
/usr/bin/ld: ogg_opus.c:(.text+0x38c): undefined reference to `opus_multistream_encoder_destroy'
/usr/bin/ld: ogg_opus.c:(.text+0x3cf): undefined reference to `opus_multistream_decoder_destroy'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_read_header':
ogg_opus.c:(.text+0xa60): undefined reference to `ogg_page_packets'
/usr/bin/ld: ogg_opus.c:(.text+0xa77): undefined reference to `ogg_page_bos'
/usr/bin/ld: ogg_opus.c:(.text+0xa97): undefined reference to `ogg_page_serialno'
/usr/bin/ld: ogg_opus.c:(.text+0xb29): undefined reference to `ogg_stream_packetout'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_setup_decoder':
ogg_opus.c:(.text+0xc3a): undefined reference to `opus_multistream_decoder_create'
/usr/bin/ld: ogg_opus.c:(.text+0xc4f): undefined reference to `opus_strerror'
/usr/bin/ld: ogg_opus.c:(.text+0xc97): undefined reference to `opus_multistream_decoder_destroy'
/usr/bin/ld: ogg_opus.c:(.text+0xd27): undefined reference to `opus_multistream_decoder_ctl'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_setup_encoder':
ogg_opus.c:(.text+0xf09): undefined reference to `opus_multistream_encoder_create'
/usr/bin/ld: ogg_opus.c:(.text+0xf7b): undefined reference to `opus_multistream_surround_encoder_create'
/usr/bin/ld: ogg_opus.c:(.text+0xf9b): undefined reference to `opus_strerror'
/usr/bin/ld: ogg_opus.c:(.text+0x1000): undefined reference to `opus_multistream_encoder_ctl'
/usr/bin/ld: ogg_opus.c:(.text+0x1044): undefined reference to `opus_multistream_encoder_ctl'
/usr/bin/ld: ogg_opus.c:(.text+0x1058): undefined reference to `opus_strerror'
/usr/bin/ld: ogg_opus.c:(.text+0x1094): undefined reference to `opus_multistream_encoder_ctl'
/usr/bin/ld: ogg_opus.c:(.text+0x10a8): undefined reference to `opus_strerror'
/usr/bin/ld: ogg_opus.c:(.text+0x1174): undefined reference to `ogg_packet_clear'
/usr/bin/ld: ogg_opus.c:(.text+0x11fe): undefined reference to `ogg_stream_init'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_write_header':
ogg_opus.c:(.text+0x1303): undefined reference to `ogg_stream_reset_serialno'
/usr/bin/ld: ogg_opus.c:(.text+0x14ff): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_opus.c:(.text+0x15a2): undefined reference to `ogg_stream_flush'
/usr/bin/ld: ogg_opus.c:(.text+0x15b4): undefined reference to `opus_get_version_string'
/usr/bin/ld: ogg_opus.c:(.text+0x15f6): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_opus.c:(.text+0x1690): undefined reference to `ogg_stream_flush'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_flush':
ogg_opus.c:(.text+0x18d0): undefined reference to `opus_multistream_encode_float'
/usr/bin/ld: ogg_opus.c:(.text+0x18e3): undefined reference to `opus_strerror'
/usr/bin/ld: ogg_opus.c:(.text+0x198b): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: ogg_opus.c:(.text+0x19c5): undefined reference to `ogg_stream_pageout'
/usr/bin/ld: ogg_opus.c:(.text+0x1a0d): undefined reference to `ogg_stream_flush'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_calculate_page_duration':
ogg_opus.c:(.text+0x1a63): undefined reference to `opus_packet_get_nb_samples'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_read_refill':
ogg_opus.c:(.text+0x1cd4): undefined reference to `opus_multistream_decode_float'
/usr/bin/ld: ogg_opus.c:(.text+0x1d06): undefined reference to `opus_packet_get_nb_samples'
/usr/bin/ld: ogg_opus.c:(.text+0x1e3c): undefined reference to `opus_multistream_decode_float'
/usr/bin/ld: ogg_opus.c:(.text+0x1e66): undefined reference to `opus_strerror'
/usr/bin/ld: ogg_opus.c:(.text+0x1eee): undefined reference to `ogg_page_eos'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_write_out':
ogg_opus.c:(.text+0x20fe): undefined reference to `opus_multistream_encoder_ctl'
/usr/bin/ld: ogg_opus.c:(.text+0x213e): undefined reference to `opus_multistream_encode_float'
/usr/bin/ld: ogg_opus.c:(.text+0x2151): undefined reference to `opus_strerror'
/usr/bin/ld: ogg_opus.c:(.text+0x228d): undefined reference to `ogg_stream_flush_fill'
/usr/bin/ld: ogg_opus.c:(.text+0x22b5): undefined reference to `ogg_stream_pageout_fill'
/usr/bin/ld: ogg_opus.c:(.text+0x233a): undefined reference to `ogg_stream_packetin'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_analyze_file':
ogg_opus.c:(.text+0x3241): undefined reference to `ogg_page_eos'
/usr/bin/ld: ogg_opus.c:(.text+0x3360): undefined reference to `ogg_page_eos'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_page_seek_search':
ogg_opus.c:(.text+0x3742): undefined reference to `opus_multistream_decoder_ctl'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_page_seek_manual':
ogg_opus.c:(.text+0x37b4): undefined reference to `ogg_stream_reset'
/usr/bin/ld: ogg_opus.c:(.text+0x3817): undefined reference to `opus_multistream_decoder_ctl'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_seek':
ogg_opus.c:(.text+0x3a7a): undefined reference to `opus_packet_get_nb_samples'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_command':
ogg_opus.c:(.text+0x3d5b): undefined reference to `opus_multistream_encoder_ctl'
/usr/bin/ld: /usr/local/lib/libsndfile.a(ogg_opus.c.o): in function `ogg_opus_byterate':
ogg_opus.c:(.text+0x402c): undefined reference to `opus_packet_get_nb_samples'
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/sutest.dir/build.make:280: sutest] Error 1
make[1]: *** [CMakeFiles/Makefile2:80: CMakeFiles/sutest.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

**Comment by BatchDrake** [comment]
This does not look a lot like a sigutils issue, but an issue in your libsndfile's install. At least, I cannot reproduce this issue locally. Could you run the following commands and copy the output here?

```
$ make clean
$ make VERBOSE=1
```

**Comment by BatchDrake** [comment]
Hi @vasagle-gleblu 

Have you managed to compile sigutils in the end?

Cheers,

**Comment by antoniovazquezblanco** [comment]
A pointer as to what may be happening...

From the line `/usr/bin/ld: /usr/local/lib/libsndfile.a` I am guessing that you have manually compiled libsndfile. This library has dependencies that seem to not have been available while compiling sigutils.

My recomendation would be to install libsndfile from mint repos (devel version included) and to remove the manually compiled file...

If you insist on manually compiling all dependencies, please make sure that all needed deps are available to the linker.

Finally, you may want to use CI deb packages, available in the development branch, avoiding all compilations. See https://github.com/BatchDrake/sigutils/actions/runs/4688680548 for example.

**Comment by antoniovazquezblanco** [comment]
@vasagle-gleblu any update on this?

**Comment by vasagle-gleblu** [comment]
Sorry, work & life getting in the way of hobbies.  
Y'know, that thing we like to collect called a paycheck.
I will look at compiling this again on the weekend.

**Comment by BatchDrake** [comment]
No worries, take your time!

**Comment by antoniovazquezblanco** [comment]
No pressure, any updates?

**Comment by vasagle-gleblu** [comment]
Believe it or not, I have a month of "mandatory" vacation coming up.


 

I will be looking at this to keep me occupied.

 

 

Sent: Wednesday, November 29, 2023 at 5:44 PM

**Comment by K7MDL2** [comment]
I just tried to do a complete new build on LinuxMint tonight, and a partial build on Win11, both from scratch having never installed/built this project before.  Initially I ran in many compile errors mostly due to missing libraries and not all of the modules were pulled from the develop branch so had mismatched code.

First to get to all the dependencies.
sndfiles is found in libsnd*
SoapySDR can be found in soapysdr*
libxml via libxm*
Qt5* for Qt framework
qtbase5-dev was required for qmake and Qt5
I could not find sutest utility on Linux build but I did on the Windows build
fftw3 via fftw3*
portaudio*
doxygen
coverxygen was not found, downloaded psycofdj/coverxygen tar file.  Read the Readme.md file
   sudo apt-get install doxygen (already done above)
   sudo apt-get install lcov
   or can run pip install coverxygen

Got lots of Compile errors in SuWidgets. mostluy about missing include files and undfined variable I, some structure errors.  After some investigation I deleted repos on my disk and started over, this time I used the git clone -b develop options for each module (sigutils, SuWidgets, suscan, SigDigger). The main page instructions leaves that out but uses the develop branch for the sigutil module.  After a good wait on my old Core i3 NUC 4 running Linux Mint I got a complete build.  I was able to run the program, success.  

Tried running it against a CaribouLite 6GHz SDR running on a Pi4B across the LAN.  I was able to hear/see a FM broadcast station.  Success.   Set to 4Mbps sample rate and 2.5MHz BW.  Lowered the Modem AGC a bit below max. Having trouble with weaker signals so digging around the settings.

On Windows had to install mingw64 and related dependencies, make and cmake stuff.  I was not able to get through the build errors there yet.

- Mike K7MDL CN87
